module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,335 @@
+                createdAt: 1
+            }
+
+            let getPollList = [
+                pollsHelper.pollDocuments
+                    (
+                        {
+                            createdBy: userId
+                        },
+                        ["name"],
+                        sortObject
+                    ),
+                database.models.pollSubmissions.aggregate
+                    (
+                        [
+                            { "$match": { "userId": userId } },
+                            {
+                                "$project": {
+                                    '_id': "$pollId",
+                                    'name': '$pollName'
+                                }
+                            },
+                            { "$sort": { createdAt: 1 } }
+                        ])
+            ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+
+                        if (response[0].length > 0) {
+                          result = result.concat(response[0]);
+                        }
+                        if (response[1].length > 0) {
+                          result = result.concat(response[1]);
+                        }
+                    });
+
+
+                if (!result.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+                
+                
+                result = result.filter(function (a) {
+                    return !this[a._id] && (this[a._id] = true);
+                }, Object.create(null));
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: result
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+
+    /**
+    * Validate poll submission.
+    * @method
+    * @name validatePollSubmission
+    * @param {String} pollId - pollId
+    * @param {String} userId - userId
+    * @returns {String} - poll status and poll submission document.
+    */
+
+   static validatePollSubmission(pollId= "", userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            if (pollId == "") {
+                throw new Error(messageConstants.apiResponses.POLL_ID_REQUIRED_CHECK)
+            }
+
+            if (userId == "") {
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+            }
+
+            let pollDocument = await pollsHelper.pollDocuments
+            (
+                { _id: pollId },
+                [
+                    "status",
+                    "endDate"
```

Comment:
@deepakr123 - We should use isDeleted filter?

Reply:
resolved

Reply:
@deepakr123 - Click on resolve buttons always.